crates/proto/src/https/https_client_stream.rs

```diff
@@ -733,11 +734,32 @@ mod tests {
         );
     }
 
-    fn client_config_tls12_webpki_roots() -> ClientConfig {
-        use rustls::{OwnedTrustAnchor, RootCertStore};
+    fn client_config_tls12() -> ClientConfig {
```

Comment:
This PR specifically only touches the loaded certificates, I didn't separate the configurations, I just left everything else as is. I don't know what the original motivation to name this function `tls12` was, I would argue that it's actually wrong. Again, this PR didn't make or name this function.

Reply:
This function is behind a `#[cfg(test)]` guard btw and is only used in these tests, maybe that caused the confusion?

Reply:
I'm aware this is the pre-existing situation. I think we should assume there wasn't a great reason (or perhaps the reason was that rustls originally required separate configs for QUIC) to have a separate 1.2/1.3 configs and so it would be nice to have a prefix commit that deduplicates this.

Reply:
Apologies, I'm not reading very carefully today, you said that from the start, I was assuming too much here :sweat_smile:!

I was just about to do that when I stumbled on the fact that unlike DoQ, DoT and DoH don't actually have a default `ClientConfig` in `trust-dns-proto`, so that's why this function is here. See [`trust_dns_proto::quic::client_config_tls13_webpki_roots`](https://docs.rs/trust-dns-proto/latest/trust_dns_proto/quic/fn.client_config_tls13_webpki_roots.html).

I could add an equivalent for DoT/DoH if you would like me to?

Reply:
So DoT and DoH get their config from the `NameServerConfigGroup`? I guess it might be better to live with the duplication in this PR and then get the configs straightened out in your next one?

Reply:
Sounds good to me!